src/ContainerInfo.cpp

```diff
@@ -72,6 +72,14 @@ std::unique_ptr<ContainerInfo> ContainerInfo::loadFromFile(
     return nullptr;
   }
 
+  std::regex matcher;
+  if (std::optional<std::string> str =
+          (*info)["matcher"].value<std::string>()) {
+    matcher = std::regex(*str, std::regex_constants::grep);
+  } else {
+    matcher = std::regex("^" + typeName, std::regex_constants::grep);
```

Comment:
Should we regex escape typeName here?

Reply:
Okay, this ended up being a bit of a rabbit hole:
- According to the comment in my C++ `bits/regex_constants.h` header:
   ```
   /**
   * Specifies that the grammar recognized by the regular expression engine is
   * that used by POSIX utility grep in IEEE Std 1003.1-2001.  This option is
   * identical to syntax_option_type basic, except that newlines are treated
   * as whitespace.
   */
   ```
- According to [IEEE Std 1003.1-2001](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=974398) (pp. 492):
  ```
  By default, an input line shall be selected if any pattern,
  treated as an entire basic regular expression (BRE) as described in the Base Definitions volume of
  IEEE Std 1003.1-2001, Section 9.3, Basic Regular Expressions, matches any part of the line
  excluding the terminating <newline>;
  ```
- According to section 9.3.3 (BRE Special Characters) (pp. 169), `. [ \ * ^ $` are the special characters. This is obviously not quite correct as the `]` is special in some contexts, but ignoring that for now.

As I understand it (though I'm struggling to find this documented) these won't collide with the C++ names that `typeName` represents, and thus shouldn't need escaping. I'm almost certain our current set are fine without escaping too.

However, if it's easy to escape I'll do it to avoid any unexpected behaviour!

Perhaps the clearer approach would be to deprecate `typeName`, as I think it's only used for this matching, and work towards everything unifying on a regex match for ultimate matching clarity.

Reply:
Ugh, I was expecting there to just be a `std::regex_escape` or similar, but I don't think I see one. In that case I think it should be fine to merge as is (or maybe panic/fail in the case we do see special characters). 

Mainly I don't want this to become a debugging rabbit hole of itself later.